### PR TITLE
drivers: nic: Add stub rtl8139 driver

### PR DESCRIPTION
**Pull request by YusufKhan-gamedev**
Adds untested, uncompiled, unanything rtl8139 driver that I made in a hurry. I dont really have plans to improve this in any capacity that would require using my brain so im just putting this PR out for someone to improve.

### CODE REVIEW

**Comment by YusufKhan-gamedev** [comment]
No, I dont know how to mark this as a draft.

**Comment by YusufKhan-gamedev** [comment]
If anyone can, add this commit and fix any compile issues that may come up

**Comment by avdgrinten** [comment]
This PR is not really *that* useful without any testing etc. If you want to try out if it actually works, feel free to sync with us on Discord or IRC.

**Comment by Dennisbonke** [comment]
Apart from the aforementioned things, I see a LGPL license specifier. Not only do we not have those kinds of license headers in the code base, I don't know if we can mix LGPL and MIT freely. Anyone that can give some insight in that?

**Comment by YusufKhan-gamedev** [comment]
> Apart from the aforementioned things, I see a LGPL license specifier. Not only do we not have those kinds of license headers in the code base, I don't know if we can mix LGPL and MIT freely. Anyone that can give some insight in that?

Pretty sure you can, see https://www.gnu.org/licenses/lgpl-3.0.en.html sections 4 and 5.

File: drivers/nic/rtl8139/src/rtl8139.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+#include <protocols/fs/server.hpp>
+#include <protocols/hw/client.hpp>
+#include <protocols/mbus/client.hpp>
+#include <protocols/svrctl/server.hpp>
+
+namespace {
+
+uint16_t readPointer;
+/* pci port address */
+auto port;
```
Comment (Dennisbonke) [inline drivers/nic/rtl8139/src/rtl8139.cpp:33]: Is auto here really needed? Same for some of the other types, I have a feeling this is a bit liberal with the auto use, which for me IMO makes the code less readable.

Comment (pitust) [reply drivers/nic/rtl8139/src/rtl8139.cpp:33]: I am pretty sure this auto is not allowed too.

Comment (YusufKhan-gamedev) [reply drivers/nic/rtl8139/src/rtl8139.cpp]: > I am pretty sure this auto is not allowed too.

Its not

**Comment by YusufKhan-gamedev** [comment]
Errors Left:

_rtl8139.cpp.o -c ../../../src/managarm/drivers/nic/rtl8139/src/rtl8139.cpp
../../../src/managarm/drivers/nic/rtl8139/src/rtl8139.cpp:125:15: error: expression is not assignable
        frame.size() =  8208;
        ~~~~~~~~~~~~ ^
../../../src/managarm/drivers/nic/rtl8139/src/rtl8139.cpp:127:20: error: expression is not assignable
        frame.byte_data() = receiveOut;
        ~~~~~~~~~~~~~~~~~ ^
2 errors generated.

Working on a different project rn(was supposed to do it yesterday) that will improve my live in the immediate time so...bye for now

**Comment by YusufKhan-gamedev** [comment]
If you want to pcik this up see https://github.com/openbsd/src/blob/master/sys/dev/ic/rtl81x9.c#L192 and fix the compile errors.

**Comment by Dennisbonke** [comment]
Yeah that is not how this works, @qookei or @avdgrinten at this point just close it, you can reopen it when you're able to do some work on it that isn't pointing us at the openbsd drivers.

**Comment by YusufKhan-gamedev** [comment]
Ah, and it doesnt update the pull request...great

**Pull request closed without merging**